studiopress/genesis-custom-blocks
# Allow opting into analytics

#### Changes
* Copies BMO's analytics work from Genesis Blocks into GCB
* Port the Settings page PHP from Pro to free
* Not in scope: using analytics, like sending an event

Fixes [GF-3267](https://wpengine.atlassian.net/browse/GF-3267)

#### Testing instructions
1. `composer i && npm i && npm run dev`
2. /wp-admin > Custom Blocks > Settings
3. Check 'Enable analytics':
![Screen Shot 2021-12-16 at 10 26 01 AM](https://user-images.githubusercontent.com/4063887/146409946-cb5cad1e-8d59-4959-bda9-e241fafd2d52.png)
4. Click 'Save Changes'
5. Create a new post
6. In the DevTools, enter `window.GcbAnalytics`
7. Expected: It's defined: 
![Screen Shot 2021-12-16 at 10 26 50 AM](https://user-images.githubusercontent.com/4063887/146410112-5b151e1c-f54f-4ba8-b6a5-58fe9f4d4f7d.png)
